common/test/acceptance/tests/lms/test_lms.py

```diff
@@ -283,6 +305,14 @@ def test_deferred_verification_enrollment(self):
         # Submit payment
         self.fake_payment_page.submit_payment()
 
+        # Expect enrollment activated event
+        assert_event_emitted_num_times(
```

Comment:
Since this event is emitted immediately after the user visits the track selection page, when they are enrolled in the course's honor mode, I'd move this up to immediately after `self.track_selection_page.visit()`, as you've done in `test_immediate_verification_enrollment`.